packages/sdk/src/types/Loyalties.ts

```diff
@@ -171,51 +482,344 @@ export interface LoyaltyFixed {
+	| ObjectCalculatePointsProportionallyCustomerMetadata
+
+interface ObjectCalculatePointsProportionallyOrderAmounts {
+	type: 'PROPORTIONAL'
+	calculation_type: 'ORDER_AMOUNT'
+	order: {
+		amount: {
+			every: number
+			points: number
+		}
+	}
+}
+
+interface ObjectCalculatePointsProportionallyOrderTotalAmounts {
+	type: 'PROPORTIONAL'
+	calculation_type: 'ORDER_TOTAL_AMOUNT'
+	order: {
+		total_amount: {
+			every: number
+			points: number
+		}
+	}
+}
+
+interface ObjectCalculatePointsProportionallyOrderMetadata {
+	type: 'PROPORTIONAL'
+	calculation_type: 'ORDER_METADATA'
+	order: {
+		metadata: {
+			every: number
+			points: number
+			property: string
+		}
+	}
+}
+
+interface ObjectCalculatePointsProportionallyItemAmount {
 	type: 'PROPORTIONAL'
-	order?: {
+	calculation_type: 'ORDER_ITEMS_AMOUNT'
+	order_items: {
 		amount: {
 			every: number
 			points: number
+			object: 'products_collection' | 'product' | 'sku'
+			id: string
 		}
 	}
 }
 
-export interface LoyaltiesEarningRulesResponse {
-	id: string
-	created_at: string
-	updated_at?: string
-	validation_rule_id?: string
-	loyalty: LoyaltyFixed | LoyaltyProportional
-	segment?: {
-		id: string
+interface ObjectCalculatePointsProportionallyItemsSubtotalAmount {
+	type: 'PROPORTIONAL'
+	calculation_type: 'ORDER_ITEMS_SUBTOTAL_AMOUNT'
+	order_items: {
+		subtotal_amount: {
+			every: number
+			points: number
+			object: 'products_collection' | 'product' | 'sku'
+			id: string
+		}
 	}
-	event: 'customer.segment.entered' | 'order.paid' | string
-	source?: {
-		banner?: string
-		object_id?: string
-		object_type?: string
+}
+
+interface ObjectCalculatePointsProportionallyItemsQuantity {
+	type: 'PROPORTIONAL'
+	calculation_type: 'ORDER_ITEMS_QUANTITY'
+	order_items: {
+		quantity: {
+			every: number
+			points: number
+			object: 'products_collection' | 'product' | 'sku'
+			id: string
+		}
 	}
-	object: 'earning_rule'
-	automation_id: string
 }
-export interface LoyaltiesListEarningRulesResponse {
-	object: 'list'
-	total: number
-	data_ref: string
-	data: LoyaltiesEarningRulesResponse[]
+
+interface ObjectCalculatePointsProportionallyCustomEventMetadata {
+	type: 'PROPORTIONAL'
+	calculation_type: 'CUSTOMER_METADATA'
+	custom_event: {
+		metadata: {
+			every: number
+			points: number
+			property: string
+		}
+	}
 }
 
-export interface LoyaltiesCreateEarningRule {
-	event: 'order.paid' | 'customer.segment.entered' | string
-	validation_rule_id?: string
-	loyalty: LoyaltyProportional | LoyaltyFixed
-	source?: { banner?: string }
-	custom_event?: { schema_id?: string }
-	segment?: { id?: string }
+interface ObjectCalculatePointsProportionallyCustomerMetadata {
+	type: 'PROPORTIONAL'
+	calculation_type: 'CUSTOMER_METADATA'
+	customer: {
+		metadata: {
+			every: number
+			points: number
+			property: string
+		}
+	}
 }
 
-export type LoyaltiesCreateEarningRuleResponse = LoyaltiesEarningRulesResponse
+export type LoyaltiesCreateEarningRule = Partial<CreateEarningRuleObject>
```

Comment:
Interface vs Type - please look at origin comment about the `DistributionsPublicationsCreateParams`